### DISCUSSION TITLE
Runtime error

### DISCUSSION THREAD

**Discussion by kumarvrai** [mdolab/pyhyp#91] [Q&A] [answered]
Hello,
After all the installations, when I run an example it produces an error:

(base) [kumarv@ilogin4 BWB]$ python runBWB.py 
Traceback (most recent call last):
  File "/gpfs/fs1/home/kumarv/pyhyp/examples/BWB/runBWB.py", line 45, in <module>
    hyp = pyHyp(options=options)
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kumarv/.local/improv/anaconda3/2024.02/lib/python3.11/site-packages/pyhyp/pyHyp.py", line 411, in __init__
    self.hyp = MExt.MExt("hyp", [curDir], debug=debug)._module
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kumarv/.local/improv/anaconda3/2024.02/lib/python3.11/site-packages/pyhyp/MExt.py", line 47, in __init__
    self._pkg = __import__(self._pkgname, globals(), locals(), [self.name])
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ImportError: /gpfs/fs1/soft/improv/software/custom-built/anaconda3/2024.2/bin/../lib/libstdc++.so.6: version `GLIBCXX_3.4.31' not found (required by /home/kumarv/packages/petsc-3.21.1/real-opt/lib/libpetsc.so.3.21)
Exception ignored in: <function MExt.__del__ at 0x15554d4b2fc0>
Traceback (most recent call last):
  File "/home/kumarv/.local/improv/anaconda3/2024.02/lib/python3.11/site-packages/pyhyp/MExt.py", line 58, in __del__
    del sys.modules[self._module.__name__]
                    ^^^^^^^^^^^^
AttributeError: 'MExt' object has no attribute '_module'
Exception ignored in: <function pyHyp.__del__ at 0x1555387d1c60>
Traceback (most recent call last):
  File "/home/kumarv/.local/improv/anaconda3/2024.02/lib/python3.11/site-packages/pyhyp/pyHyp.py", line 777, in __del__
    self.hyp.releasememory()
    ^^^^^^^^
AttributeError: 'pyHyp' object has no attribute 'hyp'

Any suggestions on how to fix this? It seems like the error is related to some kind of visualization related library, but I am not sure. I have installed PYHYP on a cluster, in case that makes any difference. 

Vishal

**Comment by marcomangano** [comment, marked as answer]
Hi Vishal, the error is related to `PETSc` . I don't know what is the issue with Anaconda and the missing `GLIBCXX`, but as you can see in the [MACH-Aero documentation](https://mdolab-mach-aero.readthedocs-hosted.com/en/latest/installInstructions/install3rdPartyPackages.html) we support up to PETSc 3.19, while you are using 3.21.
I would suggest downgrading PETSc and see if the error persist, and in case reach out to the HPC support to see if they can fix the issue with Anaconda and GLIBCXX

**Reply by kumarvrai** [reply]
Hi Marco, using different version of python (and PETSc) fixed it! Thanks for the reply.

Vishal
